kbchat/kbchat_test.go

```diff
@@ -103,82 +102,284 @@ func deleteWorkingDir(workingDir string) error {
 	return os.RemoveAll(workingDir)
 }
 
-var kbc *API
-var config Config
+type testSetupOptions struct {
+	OneOnOnePartner string
+	TeamName        string
+}
 
-func TestMain(m *testing.M) {
-	var err error
-	config, err = readAndParseConfig()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error in reading config: %v\n", err)
-	}
-	dir, err := randomTempDir()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error in generating directory: %v\n", err)
+func testSetup(botName string, options *testSetupOptions) (bot *API, config botConfig, dir string, oneOnOneChannel Channel, teamChannel Channel) {
```

Comment:
i think there's a little too much effort made here to make this be a single function. and i don't love how it's doing defaults (seems like it's not unreasonable to expect the caller to know who is in the conversation given that it's going to be doing stuff with the channel). what would this look like if calling it was either: `setupOneOnOne("alice1", "charlie1")` or `setupTeamChat("alice1", "acme")`? and then we can get rid of this testSetupOptions object which feels to me like it's not really giving enough lift for the added obfuscation.